js/lib/BTCUtils.js

```diff
@@ -0,0 +1,659 @@
+   * @param {Uint8Array}    output The output
+   * @returns {number}      The length indicated by the prefix, error if invalid length
+   */
+  determineOutputLength: (output) => {
+    let len = utils.safeSlice(output, 8, 9)[0];
+
+    if (len > 0xfd) {
+      throw new Error("Multi-byte VarInts not supported");
+    }
+
+    return BigInt(len) + 8n + 1n; // 8 byte value, 1 byte for len itself
+  },
+
+  /**
+   * @notice                Extracts the output at a given index in the TxIns vector
+   * @dev                   Iterates over the vout. If you need to extract multiple, write a custom function
+   * @param {Uint8Array}    vout The _vout to extract from
+   * @param {number}        index The 0-indexed location of the output to extract
+   * @returns {Uint8Array}  The specified output
+   */
+  extractOutputAtIndex: (vout, index) => {
+    let len;
+    let remaining;
+    let offset = 1n;
+
+    for (let i = 0; i <= index; i++) {
+      remaining = utils.safeSlice(vout, offset, vout.length - 1);
+      len = module.exports.determineOutputLength(remaining);
+      if (i !== index) {
+        offset += len;
+      }
+    }
+
+    return utils.safeSlice(vout, offset, offset + len);
+  },
+
+  /**
+   * @notice                Extracts the output script length
+   * @dev                   Indexes the length prefix on the pk_script
+   * @param {Uint8Array}    output The output
+   * @returns {Uint8Array}  The 1 byte length prefix
+   */
+  extractOutputScriptLen: (output) => {
+    return utils.safeSlice(output, 8, 9);
+  },
+
+  /**
+   * @notice                Extracts the value bytes from the output in a tx
+   * @dev                   Value is an 8-byte little-endian number
+   * @param {Uint8Array}    output The output
+   * @returns {Uint8Array}  The output value as LE bytes
+   */
+  extractValueLE: (output) => {
+    return utils.safeSlice(output, 0, 8);
+  },
+
+  /**
+   * @notice                Extracts the value from the output in a tx
+   * @dev                   Value is an 8-byte little-endian number
+   * @param {Uint8Array}    output The output
+   * @returns {Uint8Array}  The output value
+   */
+  extractValue: (output) => {
+    let leValue = module.exports.extractValueLE(output);
+    let beValue = module.exports.reverseEndianness(leValue);
+    return utils.bytesToUint(beValue);
+  },
+
+  /**
+   * @notice                Extracts the data from an op return output
+   * @dev                   Returns empty Uint8Array if no data or not an op return
+   * @param {Uint8Array}    output The output
+   * @returns {Uint8Array}  Any data contained in the opreturn output, null if not an op return
+   */
+  extractOpReturnData: (output) => {
+    if (!utils.typedArraysAreEqual(utils.safeSlice(output, 9, 10), new Uint8Array([106]))) {
+      return null;
+    }
+    let dataLen = utils.safeSlice(output, 10, 11);
+    return utils.safeSlice(output, 11, 11 + Number(utils.bytesToUint(dataLen)));
```

Comment:
there's a simpler way to do this:
```
let dataLen = output[10];
return utils.safeSlice(output, 11, 11 + dataLen);
```

Remember that u8as are arrays of `Numbers` between 0 and 255. So when we want a single element, we can just index it and treat it as a Number

Reply:
Ah, right, fixed